dipy/reconst/multi_voxel.py

```diff
@@ -3,8 +3,140 @@
 from numpy.lib.stride_tricks import as_strided
 
 from dipy.core.ndindex import ndindex
+from dipy.core.parallel import mparray_as_ndarray, ndarray_to_mparray
 from dipy.reconst.quick_squash import quick_squash as _squash
 from dipy.reconst.base import ReconstFit
+from multiprocessing import cpu_count, Pool
+
+import warnings
+import itertools
+
+
+global shared_arr
+
+
+def parallel_fit_worker(arguments):
+    """
+    Each pool process calls this worker.
+    Fit model on chunks
+
+    Parameters
+    -----------
+    arguments: tuple
+        tuple should contains a model and
+        list of indexes
+
+    Returns
+    --------
+    result: list of tuple
+        return a list of tuple(voxel index, model fitted instance)
+    """
+    model, chunks = arguments
+    return [(idx, model.fit(shared_arr[idx]))
```

Comment:
write it in a single line?

Reply:
I do not see the benefit. it will be harder to read and understand if I change that